src/Logic/Taxonomy.php

```diff
@@ -214,6 +226,148 @@ public function get_or_create_category_by_name_and_parent_id( string $cat_name,
 		return $cat_id;
 	}
 
+	/**
+	 * Gets or creates a category by its name and parent term_id.
+	 *
+	 * The array is the same as wp_insert_category accepts. https://developer.wordpress.org/reference/functions/wp_insert_category
+	 *
+	 * Note that you *have* to provide at least the 'cat_name' field.
+	 *
+	 * @param array  $data              The data to create the category with.
+	 * @param string $unique_identifier A unique identifier for your category – can be any string, but should be unique.
```

Comment:
Not sure why our PHPCS isn't picking this up, but could we update `@param string $unique_identifier` to contain possible null value, i.e. `@param ?string $unique_identifier`?

And same thing for the other method `get_or_create_tag`, please.